src/sds.c

```diff
@@ -306,15 +306,18 @@ sds sdsMakeRoomForNonGreedy(sds s, size_t addlen) {
  *
  * After the call, the passed sds string is no longer valid and all the
  * references must be substituted with the new pointer returned by the call. */
-sds sdsRemoveFreeSpace(sds s) {
-    return sdsResize(s, sdslen(s));
+sds sdsRemoveFreeSpace(sds s, int is_change_likely) {
+    return sdsResize(s, sdslen(s), is_change_likely);
 }
 
 /* Resize the allocation, this can make the allocation bigger or smaller,
  * if the size is smaller than currently used len, the data will be truncated.
+ *
+ * SDS that is likely to be changed again, won't be used with SDS_TYPE_5.
+ *
  * To avoid repeated calls, the sds allocation size will be set to the requested size
  * regardless of the actual allocation size. */
-sds sdsResize(sds s, size_t size) {
+sds sdsResize(sds s, size_t size, int is_change_likely) {
```

Comment:
`immutable_string` is a bit misleading since DB strings can still change (on APPEND, for example), so the only difference is the change likelihood between SET's strings and querybuf's string.

Reply:
@oranagra please LMK how to proceed, regarding the var name, and the module's RM_StringTruncate issue.
Thanks.

Reply:
bahh, let's keep it.. it's just an argument name for two small functions.
the worse that could happen if we find a better name, is that it'll be negated, and we'll need to fix all callers.